src/crypto/aead/aes256gcm.rs

```diff
@@ -0,0 +1,127 @@
+//! WARNING: Despite being the most popular AEAD construction due to its use in
+//! TLS, safely using AES-GCM in a different context is tricky. No more than
+//! ~ 350 GB of input data should be encrypted with a given key. This is for
+//! ~ 16 KB messages - actual figures vary according to message sizes.
+//! In addition, nonces are short and repeated nonces would totally destroy the
+//! security of this scheme. Nonces should thus come from atomic counters,
+//! which can be difficult to set up in a distributed environment.
+//! Unless you absolutely need AES-GCM, use the
+//! [default AEAD export](crate::crypto::aead)
+//! instead - it doesn't have any of these limitations. Or, if you don't need
+//! to authenticate additional data, just stick to [secretbox](crate::crypto::secretbox).
+//!
+//! AES primitives will not be made available unless your compile target
+//! is x86/x86_64 with support for the AES-NI instruction set and the CLMUL
+//! instruction (Westmere and beyond).
+
+#[cfg(any(
+    doc,
```

Comment:
Do we need the `any(doc, ...)` arm here now that the flags for `docs.rs` are adjusted? (Running `cargo doc` should produce whatever your compiler flags will produce for the, shouldn't it?)

Reply:
Unfortunately we do still need it until https://github.com/rust-lang/cargo/issues/7677 is resolved.

The docs won't build without it.